### PR TITLE
TosaToLinalgNamed: add option to prefer HWCF kernel layout for Conv2D ops.

### PR DESCRIPTION
**Pull request by bjacob**
Switching to FHWC happened in https://github.com/llvm/llvm-project/pull/68304 and is fine in itself but caused downstream performance regression https://github.com/openxla/iree/pull/15296#issuecomment-1782994965 , so this PR makes this optional.

### CODE REVIEW

**Comment by bjacob** [comment]
@FranklandJack

File: mlir/include/mlir/Conversion/TosaToLinalg/TosaToLinalg.h [hunk 1]
```diff
@@ -45,8 +48,12 @@ void registerTosaToLinalgPipelines();
 /// Populates conversion passes from TOSA dialect to Linalg dialect.
 void populateTosaToLinalgConversionPatterns(RewritePatternSet *patterns);
 
+enum class Conv2DKernelLayout { FHWC, HWCF };
+
 /// Populates conversion passes from TOSA dialect to Linalg named operations.
-void populateTosaToLinalgNamedConversionPatterns(RewritePatternSet *patterns);
+void populateTosaToLinalgNamedConversionPatterns(
```
Comment (MaheshRavishankar) [inline mlir/include/mlir/Conversion/TosaToLinalg/TosaToLinalg.h:54]: Could we make this take the `populate*` method take `TosaToLinalgNamedOptions` directly? Then we dont need the new `enum`. Also this is a entry point for *all* tosa to linalg named conversion patterns... So we want a stable ABI. Passing the options directly means that more optionality for controlling other named ops dont need API changes.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Have one small comment. Thanks!

**Review by MaheshRavishankar — APPROVED** [review]
Looks good to me.

**Pull request merged**